WalletWasabi/Wallets/TransactionFeeProvider.cs

```diff
@@ -0,0 +1,117 @@
+using NBitcoin;
```

Comment:
Added the retry mechanism, but I'm not a fan of infinite while loops, so I used a simple for loop.

Also, not a fan of `SemaphoreSlims` but added it. Whatever makes this PR mergeable.

Reply:
> Added the retry mechanism, but I'm not a fan of infinite while loops, so I used a simple for loop.

Yeah as you want

> Also, not a fan of SemaphoreSlims but added it. Whatever makes this PR mergeable.

Here it's unfortunately one of the best designs because it provides active waiting capabilities, where you were retrying each X amount of time. Your design was also incorrect for other reasons, such as if `ActionAsync` was being executed, a new request would have to wait to be scheduled until the end of the current ones to finish. So wait then wait again.